staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -743,3 +743,43 @@ func (s *Scheme) Name() string {
 // internalPackages are packages that ignored when creating a default reflector name. These packages are in the common
 // call chains to NewReflector, so they'd be low entropy names for reflectors
 var internalPackages = []string{"k8s.io/apimachinery/pkg/runtime/scheme.go"}
+
+// ToOpenAPIDefinitionName returns the REST-friendly OpenAPI definition name of the groupVersionKind.
+func (s *Scheme) ToOpenAPIDefinitionName(groupVersionKind schema.GroupVersionKind) (string, error) {
+	example, err := s.New(groupVersionKind)
+	if err != nil {
+		return "", err
+	}
+	rtype := reflect.TypeOf(example).Elem()
```

Comment:
If we're making this more official / available, do we need to protect against ending up with an unstructured.Unstructured type here?

I think scheme makes it *possible* to do things like `unstructuredScheme.AddKnownTypeWithName(gvk, &unstructured.Unstructured{})`

Reply:
Good catch.

I'd supported unstructured and documented that unstructured is assumed to follow custom resource convensions

As a sanity check, I verified that in our OpenAPI something like `/apis/stable.example.com/v1/crontabs` maps to `#/components/schemas/com.example.stable.v1.CronTab`.

Also added examples to the godoc and a couple test cases.